### PR TITLE
fix: custom column resize [WEB-1341]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Not technically a "fix" as it was just simply not supported by the original `glide-grid-table` library. The library simply chose to skip the ability to measure the cell size when it's a custom defined cell type. This PR achieves the expected behavior by doing the following:

1. fork the `glide-grid-table` into [our own](https://github.com/determined-ai/glide-data-grid) and making the changes (until `glide-grid-table` decides to accept our PR of resolving this)
2. publish the fork on [npmjs.com](https://www.npmjs.com/package/@glideapps/glide-data-grid?activeTab=code)
3. update our custom cells to provide a `measure` function to provide dynamically calculated widths for the cell

## Test Plan

- navigate to experiment list view `?f_explist_v2=on` query param if still applicable
- add a variety of different cell types via the column picker
  - link cells (exp name)
  - user avatar cells (user)
  - tags cells (tags)
  - checkbox cells (multi-select checkboxes)
  - experiment stage cells (experiment state)
- double click on the right edge of the column header for each respective column
- verify that the column dynamically resizes to fit the loaded cell content (like how spreadsheets do it)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-1341

### CODE REVIEW

**Comment by ashtonG** [comment]
quick q: why do we add the middle mouse button to the mousedown and mouseup handlers?

**Review by alexjdetermined — APPROVED** [review]
LGTM

File: webui/react/package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@codemirror/lang-markdown": "^6.1.1",
     "@codemirror/lang-python": "^6.1.2",
     "@codemirror/legacy-modes": "^6.3.2",
-    "@glideapps/glide-data-grid": "^5.2.1",
+    "@hpe.com/glide-data-grid": "^5.2.1",
```
Comment (keita-determined) [inline webui/react/package.json:37]: i think its nice to have docs how to maintain the repo in confluence

**Review by keita-determined — COMMENTED** [review]
it mostly works but text is chopped off in `description` and `tag` columns when text contents are long
This is not expected, right? I think its not `like how spreadsheets do it` because spreadsheet shows the entire text contents

https://github.com/determined-ai/determined/assets/109113616/31dabd60-b8c5-450b-8532-c99743a373c2

**Comment by hkang1** [comment]
> it mostly works but text is chopped off in `description` and `tag` columns when text contents are long This is not expected, right? I think its not `like how spreadsheets do it` because spreadsheet shows the entire text contents. Or is it acceptable?
> 
>  Screen.Recording.2023-07-17.at.2.04.11.PM.mov

I think the cells in the sheets truncate the content when the given space is not enough. I guess another alternative behavior would be to show an ellipsis. What kind of behavior were you thinking of?

File: webui/react/package.json (hunk 1 above)
Comment (hkang1) [reply webui/react/package.json:37]: great idea

**Comment by keita-determined** [comment]
> I think the cells in the sheets truncate the content when the given space is not enough. I guess another alternative behavior would be to show an ellipsis. What kind of behavior were you thinking of?

I expected to show all text like spreadsheet but i noticed that there are min/max width for each column. I thought the width could be unlimited

**Review by keita-determined — APPROVED** [review]
Did we agree to use `@hpe.com/glide-data-grid`? If so, LGTM

**Comment by hkang1** [comment]
> Did we agree to use `@hpe.com/glide-data-grid`? If so, LGTM

Yeah this was the only way for now to get the resize behavior for now. There is an open PR on the original `glide-data-grid` to bring this into the original framework but the developers there have been busy.

File: webui/react/package.json (hunk 1 above)
Comment (hkang1) [reply webui/react/package.json:37]: added forked entry explanation in our wiki

**Pull request merged**